<?xml version="1.0" encoding="UTF-8"?>
<!--
   Copyright 2011 Research In Motion Limited.

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.

-->

<ui version="4.0">
 <class>MemoryDialog</class>
 <widget class="QDialog" name="MemoryDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>583</width>
    <height>356</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="lblCurrent">
       <property name="text">
        <string>Current</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QLabel" name="lblHighWatermark">
       <property name="text">
        <string>High Watermark</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="lblMemoryUsed">
       <property name="text">
        <string>Total Memory Used</string>
       </property>
       <property name="buddy">
        <cstring>currentTotalMemoryUsed</cstring>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="currentTotalMemoryUsed">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLineEdit" name="highTotalMemoryUsed">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="lblPageCacheSize">
       <property name="text">
        <string>Page Cache Size</string>
       </property>
       <property name="buddy">
        <cstring>currentPageCacheSize</cstring>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="currentPageCacheSize">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QLineEdit" name="highPageCacheSize">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="lblPageCachedUsed">
       <property name="text">
        <string>Page Cache Used</string>
       </property>
       <property name="buddy">
        <cstring>currentPageCacheUsed</cstring>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="currentPageCacheUsed">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="3" column="2">
      <widget class="QLineEdit" name="highPageCacheUsed">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="lblCacheOverflow">
       <property name="text">
        <string>Page Cache Overflow</string>
       </property>
       <property name="buddy">
        <cstring>currentPageCacheOverflow</cstring>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLineEdit" name="currentPageCacheOverflow">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="4" column="2">
      <widget class="QLineEdit" name="highPageCacheOverflow">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="lblParserStack">
       <property name="text">
        <string>Parser Stack</string>
       </property>
       <property name="buddy">
        <cstring>currentParserStack</cstring>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QLineEdit" name="currentParserStack">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="5" column="2">
      <widget class="QLineEdit" name="highParserStack">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="lblMallocCount">
       <property name="text">
        <string>Malloc Count</string>
       </property>
       <property name="buddy">
        <cstring>currentMallocCount</cstring>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QLineEdit" name="currentMallocCount">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="6" column="2">
      <widget class="QLineEdit" name="highMallocCount">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="7" column="0">
      <widget class="QLabel" name="lblMallocSize">
       <property name="text">
        <string>Malloc Size</string>
       </property>
       <property name="buddy">
        <cstring>currentMallocSize</cstring>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QLineEdit" name="currentMallocSize">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="7" column="2">
      <widget class="QLineEdit" name="highMallocSize">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>96</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="resetHighWatermarkButton">
       <property name="text">
        <string>Reset High Watermark</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="closeButton">
       <property name="text">
        <string>Close</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>currentTotalMemoryUsed</tabstop>
  <tabstop>highTotalMemoryUsed</tabstop>
  <tabstop>currentPageCacheSize</tabstop>
  <tabstop>highPageCacheSize</tabstop>
  <tabstop>currentPageCacheUsed</tabstop>
  <tabstop>highPageCacheUsed</tabstop>
  <tabstop>currentPageCacheOverflow</tabstop>
  <tabstop>highPageCacheOverflow</tabstop>
  <tabstop>currentParserStack</tabstop>
  <tabstop>highParserStack</tabstop>
  <tabstop>currentMallocCount</tabstop>
  <tabstop>highMallocCount</tabstop>
  <tabstop>currentMallocSize</tabstop>
  <tabstop>highMallocSize</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>closeButton</sender>
   <signal>clicked()</signal>
   <receiver>MemoryDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>406</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
